File: ble-common/src/main/java/no/nordicsemi/android/ble/common/profile/rsc/RunningSpeedAndCadenceMeasurementCallback.java
package no.nordicsemi.android.ble.common.profile.rsc;

import android.bluetooth.BluetoothDevice;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

@SuppressWarnings("unused")
public interface RunningSpeedAndCadenceMeasurementCallback {

	/**
	 * Method called when the Running Speed And Cadence Measurement data was received.
	 *
	 * @param device               the target device.
	 * @param running              true if running was detected, false if walking.
	 * @param instantaneousSpeed   instantaneous speed in m/s unit.
	 * @param instantaneousCadence instantaneous cadence in steps per minute.
	 * @param strideLength         instantaneous stride length in centimeters.
	 * @param totalDistance        the total distance in meters (UINT32).
	 */
	void onRSCMeasurementReceived(@NonNull final BluetoothDevice device, final boolean running,
								  final float instantaneousSpeed, final int instantaneousCadence,
								  @Nullable final Integer strideLength,
								  @Nullable final Long totalDistance);
}

File: ble-common/src/main/java/no/nordicsemi/android/ble/common/profile/rsc/RunningSpeedAndCadenceMeasurementCallback.kt
/*
 * Copyright (c) 2018, Nordic Semiconductor
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
 *
 * 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
 *
 * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the
 * documentation and/or other materials provided with the distribution.
 *
 * 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this
 * software without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
 * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
 * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
 * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
 * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 * USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */

package no.nordicsemi.android.ble.common.profile.rsc

import android.bluetooth.BluetoothDevice

interface RunningSpeedAndCadenceMeasurementCallback {

    /**
     * Method called when the Running Speed And Cadence Measurement data was received.
     *
     * @param device               the target device.
     * @param running              true if running was detected, false if walking.
     * @param instantaneousSpeed   instantaneous speed in m/s unit.
     * @param instantaneousCadence instantaneous cadence in steps per minute.
     * @param strideLength         instantaneous stride length in centimeters.
     * @param totalDistance        the total distance in meters (UINT32).
     */
    fun onRSCMeasurementReceived(
        device: BluetoothDevice, running: Boolean,
        instantaneousSpeed: Float, instantaneousCadence: Int,
        strideLength: Int?,
        totalDistance: Long?
    )
}
